lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1320,7 +1325,16 @@ public FoundHits searchObjects(String objectType, MatchFilter matchFilter,
     private List<Map<String, Object>> prepareMatchFilters(MatchFilter matchFilter) {
         List<Map<String, Object>> ret = new ArrayList<>();
         if (matchFilter.fullTextInAll != null) {
-            ret.add(createFilter("match", "_all", matchFilter.fullTextInAll));
```

Comment:
Just to be clear, because I still don't understand - should the create filter method be modified with this new behavior?

Reply:
Or should createFilter be modified to support both the old and new behavior, controlled by a boolean?

Reply:
my impression was that the change was to get ride of the old behavior entirely

Reply:
So the createFilter method should be removed then?

Reply:
actually createFilter should be renamed (not sure what maybe HelpMeNestThingsSingularly). Its used everywhere as helper to nest commands. For ex. updateAccessGroupForVersions, setFieldOnObject and a bunch of methods use it. These methods make post request and so is not what I am looking to change.... 

None of the search methods directly call it (only through prepareMatchFilters) and the fullTextInAll field is the only place I need to double nest things. objectName don't have spaces and lookupInKeys uses another type of query that defaults to "AND" behavior already.

Reply:
So now what I'm hearing is that createFilter's behavior should stay as is (with a possible fn rename to better state what the fn does) and the new behavior you're adding is only relevant in the one place it currently exists?

Reply:
yes. It actually makes more sense to rename fn prepareMatchfilters as "prepareFilter"(or "createFilters") s cuz that is what it is doing. Esp since elasticsearch call their filters filters and not matchfilters... The current fn createFilters is more like add all kinds of key/value pairs, not just filters, to calls